crates/starknet/src/starknet/mod.rs

```diff
@@ -252,16 +252,44 @@ impl Starknet {
         Ok(())
     }
 
-    // TODO should return a more generic type (StateReader) to allow future implementation of a
-    // ForkedStateReader
-    pub fn get_state_reader_at(&self, block_id: &BlockId) -> Result<&InMemoryStateReader> {
+    fn get_state_at(&self, block_id: &BlockId) -> Result<&StarknetState> {
         match block_id {
-            BlockId::Tag(_) => Ok(&self.state.state),
+            BlockId::Tag(_) => Ok(&self.state),
             BlockId::Hash(_) => Err(Error::BlockIdHashUnimplementedError),
             BlockId::Number(_) => Err(Error::BlockIdNumberUnimplementedError),
         }
     }
 
+    pub fn get_class_hash_at(
+        &self,
+        block_id: &BlockId,
+        contract_address: &Address,
+    ) -> Result<Felt> {
+        let state = self.get_state_at(block_id)?;
+        match state.state.address_to_class_hash.get(contract_address) {
+            Some(class_hash) => Ok(Felt::from(*class_hash)),
+            None => Err(Error::ContractNotFound),
+        }
+    }
+
+    pub fn call(&self, block_id: BlockId, function_call: FunctionCall) -> Result<Vec<Felt>> {
```

Comment:
I see that FunctionCall is having 3 properties, each one of them is Felt which is already present in types crate and converts via From trait to Felt252, so is it going to be easier to just accept contract_address, entry_point_selector and calldata using the already defined types in the types crate and then internally convert to starknet_in_rust types. That way maybe the `From<FunctionCall> for starknet_rs_core::FunctionCall` will not be needed

Reply:
Addressed in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/76/commits/8dae79c7a3105446869ccc667596cd3f89beb0ff

Reply:
Removed unused `From<FunctionCall>` in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/76/commits/250eabdf049c3fb04427457cfa7a4e33463167c4